### PR TITLE
Get installed dotnet runtimes

### PR DESCRIPTION
**Pull request by vbjay**
## Proposed changes

- Add dotnet version info to the copy output

### Before

- Git Extensions 4.0.0.15569
- Build cba315df5366fca01240d93b6e5a5077fd5e6630
- Git 2.38.1.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.11
- DPI 96dpi (no scaling)

### After
- Git Extensions 33.33.33
- Build 1401bc1c797431b64b4ce85be5b9ae79d8161be0
- Git 2.38.1.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.11
- DPI 96dpi (no scaling)
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 3.1.30 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

## Test methodology 

- Debugged and copied output.
- changed command name to dotnett and tested error flow

## Test environment(s) 

- GIT 
- Windows 

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/UserEnvironmentInformation.cs [hunk 1]
```diff
@@ -67,6 +69,48 @@ public static string GetGitVersionInfo(string? gitVersion, GitVersion lastSuppor
...
+                new ArgumentBuilder()
+                {
+                    "--list-runtimes"
+                },
+
+                new ArgumentBuilder()
+                {
+                    "--list-sdks"
+                }
+           };
```
Comment (mstv) [inline GitUI/UserEnvironmentInformation.cs:87]: nit: indent; `string[]` would have worked, too, wouldn't it?

Comment (vbjay) [reply GitUI/UserEnvironmentInformation.cs:87]: Yes but they are arguments and not strictly string arrays and I'm making it so if there is another command with multiple arguments, it's easier to manage and update.

File: GitUI/UserEnvironmentInformation.cs [hunk 2]
```diff
@@ -67,6 +69,48 @@ public static string GetGitVersionInfo(string? gitVersion, GitVersion lastSuppor
...
+            ArgumentString[] args = new ArgumentString[]
+           {
+                new ArgumentBuilder()
+                {
+                    "--list-runtimes"
+                },
+
+                new ArgumentBuilder()
+                {
+                    "--list-sdks"
```
Comment (RussKie) [inline GitUI/UserEnvironmentInformation.cs:85]: Why do we need the SDK information? Only runtime info is relevant.

Comment (RussKie) [reply GitUI/UserEnvironmentInformation.cs:85]: ...and arguably only Microsoft.WindowsDesktop.App is important.
We need to conscious of the length of the querysting which is passed when "create GitHub" issue functionality is invoked.

**Comment by RussKie** [comment]
image

What's your core.autocrlf setting?

File: GitUI/UserEnvironmentInformation.cs (hunk 2 above)
Comment (vbjay) [reply GitUI/UserEnvironmentInformation.cs:85]: Ok. I could regex the desktop app lines. Along with not doing sdk. Sdk was to catch any odities between runtimes vs sdks.

**Comment by RussKie** [comment]
SDK is only required to build apps, here we're interested in knowing what's
used to run Git Extensions.

**Comment by vbjay** [comment]
> image
> 
> What's your core.autocrlf setting?

image

**Review by mstv — APPROVED** [review]

**Comment by vbjay** [comment]
```bash
git rev-parse HEAD >info.txt
git ls-files --eol /mnt/c/git/gitextensions/GitUI/UserEnvironmentInformation.cs >>info.txt
```
```
62f36913740c6f9738315d892e2f34338425951a
i/lf    w/crlf  attr/text             	GitUI/UserEnvironmentInformation.cs
```

**Review by gerhardol — APPROVED** [review]

File: GitUI/UserEnvironmentInformation.cs [hunk 3]
```diff
@@ -67,6 +69,40 @@ public static string GetGitVersionInfo(string? gitVersion, GitVersion lastSuppor
             return gitVersion;
         }
 
+        public static string GetDotnetVersionInfo()
```
Comment (RussKie) [inline GitUI/UserEnvironmentInformation.cs:72]:
```suggestion
        private static string GetDotnetVersionInfo()
```

**Review by RussKie — COMMENTED** [review]
:+1: with the suggestion.

**Pull request merged**